chef-config/lib/chef-config/config.rb

```diff
@@ -29,7 +29,7 @@
 require_relative "path_helper"
 require_relative "mixin/fuzzy_hostname_matcher"
 
-Mixlib.autoload :Shellout, "mixlib/shellout"
+Mixlib.autoload :ShellOut, "mixlib/shellout"
```

Comment:
Oh right, nested class constants...  There's gotta be a better syntax to use here...

Reply:
(it appears there isn't)

Reply:
I've been poking around and it looks like a better convention is:
```
module Mixlib
    autoload :ShellOut, "mixlib/shellout"
end
```
It srill feels clumsy but it is more precise for the cases where the module` may not be loaded.

Reply:
yeah that would be my concern, and yeah that would be my suggestion.

(also means you have to know if its a module or a class at the top level, which is mildly annoying)

Reply:
(although that may only matter for `class Chef` and everything else should be properly a module I hope)